app/controllers/api/v1/contacts_controller.rb

```diff
@@ -4,12 +4,37 @@ class ContactsController < ApplicationController
       before_action :authenticate_user
 
       def index
-				authorize Contact
```

Comment:
A lot of thought and work this refractor. Super precise way for this specific action. 

However lemme be the Devil's advocate: I could potentially see this logic moved into the model in the future. Then those methods called here. As of now, it bulks the controller.